PopPUNK/assign.py

```diff
@@ -111,18 +120,19 @@ def assign_query(dbFuncs,
                 rNames.append(reference.rstrip())
     else:
         rNames = getSeqsInDb(ref_db + "/" + os.path.basename(ref_db) + ".h5")
-
-    # construct database
-    createDatabaseDir(output, kmers)
-    qNames = constructDatabase(q_files,
-                                kmers,
-                                sketch_sizes,
-                                output,
-                                threads,
-                                overwrite,
-                                codon_phased = codon_phased,
-                                calc_random = False)
-
+    if not web:
+        # construct database
+        createDatabaseDir(output, kmers)
+        qNames = constructDatabase(q_files,
+                                    kmers,
+                                    sketch_sizes,
+                                    output,
+                                    threads,
+                                    overwrite,
+                                    codon_phased = codon_phased,
+                                    calc_random = False)
+    else:
+        qNames = sketch_to_hdf5(sketch, output)
```

Comment:
It probably makes sense to add a section at the end which removes the output HDF5 file once complete